src/scanner/mod.rs

```diff
@@ -191,6 +203,90 @@ impl Scanner {
         .await?;
         Ok(stream)
     }
+
+    /// Binds to a UDP socket so we can send and recieve packets
+    /// # Example
+    ///
+    /// ```compile_fail
+    /// # use std::net::{IpAddr, Ipv6Addr, SocketAddr};
+    /// let port: u16 = 80;
+    /// // ip is an IpAddr type
+    /// let ip = IpAddr::V6(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 1));
+    /// let socket = SocketAddr::new(ip, port);
+    /// scanner.udp_bind(socket);
+    /// // returns Result which is either Ok(stream) for port is open, or Er for port is closed.
+    /// // Timeout occurs after self.timeout seconds
+    /// ```
+    ///
+    async fn udp_bind(&self, socket: SocketAddr) -> io::Result<UdpSocket> {
+        let local_addr = match socket {
+            SocketAddr::V4(_) => "0.0.0.0:0".parse::<SocketAddr>().unwrap(),
+            SocketAddr::V6(_) => "[::]:0".parse::<SocketAddr>().unwrap(),
+        };
+
+        UdpSocket::bind(local_addr).await
+    }
+
+    /// Performs a UDP scan on the specified socket with a payload and wait duration
+    /// # Example
+    ///
+    /// ```compile_fail
+    /// # use std::net::{IpAddr, Ipv6Addr, SocketAddr};
+    /// # use std::time::Duration;
+    /// let port: u16 = 123;
+    /// // ip is an IpAddr type
+    /// let ip = IpAddr::V6(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 1));
+    /// let socket = SocketAddr::new(ip, port);
+    /// let payload = vec![0, 1, 2, 3];
+    /// let wait = Duration::from_secs(1);
+    /// let result = scanner.udp_scan(socket, payload, wait).await;
+    /// // returns Result which is either Ok(true) if response received, or Ok(false) if timed out.
+    /// // Err is returned for other I/O errors.
+    async fn udp_scan(
+        &self,
+        socket: SocketAddr,
+        payload: &[u8],
+        wait: Duration,
+    ) -> io::Result<bool> {
+        match self.udp_bind(socket).await {
+            Ok(x) => {
+                let mut buf = [0u8; 1024];
+
+                x.connect(socket).await?;
+                x.send(&payload).await?;
+
+                match io::timeout(wait, x.recv(&mut buf)).await {
+                    Ok(size) => {
+                        println!("Recived {} bytes", size);
```

Comment:
suggestion: use the `debug!` macro instead of `println!` throughout.

Probably better to use the already-imported `debug!` macro (or another level, if appropriate), rather than `println!` throughout, so it can filtered.

Reply:
Additionally, typo. in `Recived`.

Reply:
@BrendanGlancy, I can see this has been updated—thanks—but there's still a typo. in `Recieved`, I'm afraid.